packages/components/search-expandable/search-expandable.ts

```diff
@@ -0,0 +1,207 @@
+import { ControlValueAccessor, FormsModule, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { KbqButton, KbqButtonModule } from '@koobiq/components/button';
+import { KBQ_LOCALE_SERVICE, ruRULocaleData } from '@koobiq/components/core';
+import { KbqFormFieldModule } from '@koobiq/components/form-field';
+import { KbqIconModule } from '@koobiq/components/icon';
+import { KbqInput, KbqInputModule } from '@koobiq/components/input';
+import { KbqToolTipModule, KbqTooltipTrigger } from '@koobiq/components/tooltip';
+import { BehaviorSubject, debounceTime, distinctUntilChanged, filter } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+/** default configuration of search-expandable */
+export const KBQ_SEARCH_EXPANDABLE_DEFAULT_CONFIGURATION = ruRULocaleData.searchExpandable;
+
+/** Injection Token for providing configuration of search-expandable */
+export const KBQ_SEARCH_EXPANDABLE_CONFIGURATION = new InjectionToken('KbqSearchExpandableConfiguration');
+
+export const defaultValue = '';
+export const defaultEmitValueTimeout = 200;
+
+@Component({
+    standalone: true,
+    selector: 'kbq-search-expandable',
+    templateUrl: './search-expandable.html',
+    styleUrls: ['./search-expandable.scss'],
+    host: {
+        class: 'kbq-search-expandable',
+        '[class.kbq-search-expandable_opened]': 'isOpened'
+    },
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    imports: [
+        KbqButtonModule,
+        KbqIconModule,
+        KbqInputModule,
+        FormsModule,
+        AsyncPipe,
+        KbqToolTipModule,
+        KbqFormFieldModule
+    ],
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: KbqSearchExpandable,
+            multi: true
+        }
+    ]
+})
+export class KbqSearchExpandable implements ControlValueAccessor, AfterViewInit {
+    /** @docs-private */
+    protected readonly localeService = inject(KBQ_LOCALE_SERVICE, { optional: true });
+    /** @docs-private */
+    protected readonly destroyRef = inject(DestroyRef);
+    /** @docs-private */
+    protected readonly changeDetectorRef = inject(ChangeDetectorRef);
+
+    readonly externalConfiguration = inject(KBQ_SEARCH_EXPANDABLE_CONFIGURATION, { optional: true });
+
+    @ViewChildren(KbqInput) private input: QueryList<KbqInput>;
+    @ViewChildren(KbqButton) private button: QueryList<KbqButton>;
+    @ViewChild(KbqTooltipTrigger) private tooltip: KbqTooltipTrigger;
+
+    configuration;
+
+    /** Current value in input. */
+    value = new BehaviorSubject(defaultValue);
+
+    /** state of component. */
+    @Input({ transform: booleanAttribute }) isOpened = false;
+    /** Emit event by enter or not. Default is false */
+    @Input() isEmitValueByEnterEnabled = false;
+    /** Timeout in milliseconds for emit event. The default value is taken from defaultEmitValueTimeout */
+    @Input() emitValueTimeout = defaultEmitValueTimeout;
```

Comment:
```suggestion
    @Input({ transform: numberAttribute }) emitValueDebounceTime = defaultEmitValueDebounceTime;
```

давай добавим трансформ для удобства

кажется, что в имени переменной лучше использовать `DebounceTime` вместо `Timeout`, тк таймаут предполагает просто задержку перед выполнением

Reply:
это именование из инфосека.

трансформ да, добавлю.